File: packages/lector/test/fixtures/python-reference/history/v2/payment.py
from abc import ABC, abstractmethod


class PurchaseOrder:
    def __init__(self, amount: int, currency: str) -> None:
        self.amount = amount
        self.currency = currency


class Receipt:
    def __init__(self, order: PurchaseOrder, processed: bool) -> None:
        self.order = order
        self.processed = processed


class PaymentProcessor(ABC):
    @abstractmethod
    def process(self, order: PurchaseOrder) -> Receipt: ...

File: packages/lector/test/fixtures/rust-reference/history/v2/payment.rs
#[derive(Clone)]
pub struct PurchaseOrder {
    pub amount: i64,
    pub currency: String,
}

pub struct Receipt {
    pub order: PurchaseOrder,
    pub processed: bool,
}

pub trait PaymentProcessor {
    fn process(&self, order: PurchaseOrder) -> Receipt;
}
